internal/provider/resource_stack_variable_set.go

```diff
@@ -0,0 +1,244 @@
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &resourceStackVariableSet{}
+var _ resource.ResourceWithConfigure = &resourceStackVariableSet{}
+var _ resource.ResourceWithImportState = &resourceStackVariableSet{}
+
+func NewStackVariableSetResource() resource.Resource {
+	return &resourceStackVariableSet{}
+}
+
+// resourceStackVariableSet implements the tfe_stack_variable_set resource type
+type resourceStackVariableSet struct {
+	config ConfiguredClient
+}
+
+// Model describes the resource data model.
+type stackVariableSetResourceModel struct {
+	ID            types.String `tfsdk:"id"`
+	StackID       types.String `tfsdk:"stack_id"`
+	VariableSetID types.String `tfsdk:"variable_set_id"`
+}
+
+func (r *resourceStackVariableSet) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_stack_variable_set"
+}
+
+func (r *resourceStackVariableSet) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages associations between variable sets and stacks.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the stack variable set association.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"variable_set_id": schema.StringAttribute{
+				Description: "The ID of the variable set.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(), // Follows tfe_workspace_variable_set and tfe_project_variable_set pattern
+				},
+			},
+			"stack_id": schema.StringAttribute{
+				Description: "The ID of the stack.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(), // Follows tfe_workspace_variable_set and tfe_project_variable_set pattern
+				},
+			},
+		},
+	}
+}
+
+func (r *resourceStackVariableSet) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(ConfiguredClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected ConfiguredClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.config = client
+}
+
+func (r *resourceStackVariableSet) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan stackVariableSetResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	vSID := plan.VariableSetID.ValueString()
+	stkID := plan.StackID.ValueString()
```

Comment:
nit: Consider the full expanded names for readability:
```suggestion
	variableSetID := plan.VariableSetID.ValueString()
	stackID := plan.StackID.ValueString()
```